src/hammer-tech/stackup.py

```diff
@@ -0,0 +1,340 @@
+
+    def max_width_from_pitch(self, pitch: float) -> float:
+        """
+        Derive the maximum wire width for a maximally-sized wire given a
+        desired pitch.
+        See min_spacing_and_max_width_from_pitch for details.
+
+        :param pitch: Desired pitch
+        :return: Maximum wire width for said pitch.
+        """
+        return self.min_spacing_and_max_width_from_pitch(pitch)[1]
+
+    def get_width_spacing_start_twt(self, tracks: int) -> Tuple[float, float, float]:
+        """
+        This method will return the maximum width a wire can be in order
+        to consume a given number of routing tracks.
+        This assumes the neighbors of the wide wire are minimum-width routes.
+        i.e. T W T
+        T = thin / min-width
+        W = wide
+        See min_spacing_and_max_width_from_pitch for an explanation of the calculation.
+
+        :param tracks: Number of routing tracks to consume
+        :return: Returns tuple of (width, spacing, start)
+        """
+        widths_and_spacings = self.power_strap_widths_and_spacings
+        spacing = widths_and_spacings[0].min_spacing
+        # the T W T pattern contains one wires (W) and 2 spaces (S2)
+        s2w = self.snap((tracks + 1) * self.pitch - self.min_width)
+
+        assert int(self.snap(s2w / self.grid_unit)) % 2 == 0, "This calculation should always produce an even s2w"
+
+        width = self.snap(s2w - spacing*2)
+        for first, second in zip(widths_and_spacings[:-1], widths_and_spacings[1:]):
+            if s2w >= second.min_spacing*2 + second.width_at_least:
+                spacing = second.min_spacing
+                width = self.snap(s2w - spacing*2)
+            elif s2w >= first.min_spacing*2 + second.width_at_least:
+                # we are asking for a pitch that is width-constrained
+                if int(second.width_at_least / self.grid_unit) % 2 == 0:
+                    # even
+                    width = self.snap(second.width_at_least - (self.grid_unit*2))
+                else:
+                    # odd
+                    width = self.snap(second.width_at_least - self.grid_unit)
+                spacing = self.snap((s2w - width)/2.0)
+
+        assert int(self.snap(self.min_width / self.grid_unit)) % 2 == 0, (
+            "Assuming all min widths are even here, if not fix me")
+        assert int(self.snap(width / self.grid_unit)) % 2 == 0, (
+            "This calculation should always produce an even width")
+
+        start = self.snap(self.min_width / 2.0 + spacing)
+        return (width, spacing, start)
+
+    def get_width_spacing_start_twwt(self, tracks: int, force_even: bool = False) -> Tuple[float, float, float]:
+        """
+        This method will return the maximum width a wire can be in order
+        to consume a given number of routing tracks.
+        This assumes that the wires are in the following configuration.
+        TODO: double check that this pydoc is correct
```

Comment:
@jwright6323 If it's easy for you to do so, would you mind helping me double check this pydoc and fill out the documentation for `force_even` below?

Reply:
done.